legate/cpp/tile_io.cpp

```diff
@@ -105,6 +106,71 @@ struct tile_read_write_fn {
   }
 };
 
+/**
+ * @brief Flatten the domain point to a 1D point
+ *
+ * @param lo_dp Lower point
+ * @param hi_dp High point
+ * @param point_dp The domain point to flatten
+ * @return The flatten domain point
+ */
+template <int32_t DIM>
+size_t linearize(const legate::DomainPoint& lo_dp,
+                 const legate::DomainPoint& hi_dp,
+                 const legate::DomainPoint& point_dp)
+{
+  const legate::Point<DIM> lo      = lo_dp;
+  const legate::Point<DIM> hi      = hi_dp;
+  const legate::Point<DIM> point   = point_dp - lo_dp;
+  const legate::Point<DIM> extents = hi - lo + legate::Point<DIM>::ONES();
+  size_t idx                       = 0;
+  for (int32_t dim = 0; dim < DIM; ++dim) {
+    idx = idx * extents[dim] + point[dim];
+  }
+  return idx;
+}
+
+/**
+ * @brief Functor for tiling read Legate store by offsets from disk using KvikIO
+ *
+ * @param context The Legate task context
+ * @param store The Legate output store
+ */
+struct tile_read_by_offsets_fn {
+  template <legate::Type::Code CODE, int32_t DIM>
+  void operator()(legate::TaskContext& context, legate::Store& store)
+  {
+    using DTYPE                             = legate::legate_type_of<CODE>;
+    const auto task_index                   = context.get_task_index();
+    const auto launch_domain                = context.get_launch_domain();
+    const std::string path                  = context.scalars().at(0).value<std::string>();
+    legate::Span<const uint64_t> offsets    = context.scalars().at(1).values<uint64_t>();
+    legate::Span<const uint64_t> sizes      = context.scalars().at(2).values<uint64_t>();
+    legate::Span<const uint64_t> tile_shape = context.scalars().at(3).values<uint64_t>();
+
+    // Flatten task index
+    uint32_t flatten_task_index = 0;
+    if (!context.is_single_task()) {
+      flatten_task_index = linearize<DIM>(launch_domain.lo(), launch_domain.hi(), task_index);
+    }
```

Comment:
Why do you need to special-case the single task case here? Or is this a micro-optimisation?

Reply:
Right, it isn't needed anymore. Removed